gems/pending/postgres_ha_admin/postgres_ha_admin.rb

```diff
@@ -0,0 +1,16 @@
+require 'util/postgres_dsn_parser'
+
+module PostgresHaAdmin
+  DB_YML_FILE = 'failover_databases.yml'.freeze
+  attr_reader :yml_file, :connection_hash, :log, :connection_hash
+
+  def init_postgres_ha_admin(config_dir, log_dir, connection_params_hash)
+    @yml_file = Pathname.new(config_dir).join(DB_YML_FILE)
+    @log = PostgresHaLogger.new(log_dir)
+    @connection_hash = connection_params_hash
+  end
+  
+  def pg_connection
+    PG::Connection.open(connection_hash)
+  end
```

Comment:
It feels like all of this except the constant should probably be in the `FailoverDatabases` class.
I'm not sure what we gain from this being here instead as `init_postgres_ha_admin` is only called from that class and you can probably just use `PG::Connection.open(connection_hash)` in place of the `pg_connection` method.